src/shutdown.h

```diff
@@ -19,10 +19,12 @@ void powerCommandsCheck()
         NVIC_SystemReset();
 #elif defined(ARCH_RP2040)
         rp2040.reboot();
-#elif defined(ARCH_RASPBERRY_PI)
+#elif defined(ARCH_RASPBERRY_PI) || defined(ARCH_PORTDUINO)
         deInitApiServer();
+#if defined(ARCH_RASPBERRY_PI)
         if (aLinuxInputImpl)
             aLinuxInputImpl->deInit();
+#endif
         SPI.end();
         Wire.end();
         Serial1.end();
```

Comment:
After digging further into my issue, I found that it's caused by the call to `Serial1.end()`. Since this is only used for GPS on `ARCH_RASPBERRY_PI` and not for `ARCH_PORTDUINO`, its file descriptor is still 0, which is normally used for `stdin`. Thus when rebooting, `stdin` is closed and after the reboot, Linux sees 0 as free and assigns the socket to it. 

@wnagele can you therefore move `Serial1.end()` into the `#if defined(ARCH_RASPBERRY_PI)`? This fixes my issue.

Reply:
I've given the `ARCH_PORTDUINO` its own block now. Otherwise the macros become unwieldy and hard to read.